eclipse-che/che-server
# Propagate Che user informations and preferences to DevWorkspaces

### What does this PR do?
Creates secrets of user profile and preferences in user namespace. 
These secrets can then be mounted by devworkspace controller into workspace containers.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19061

### How to test this PR?

1. Deploy image with this PR.
2. Create workspace from samples.
3. There should be `user-profile` and `user-preferences` secrets in your user's namespace.
![image](https://user-images.githubusercontent.com/73115616/135254336-30c6810f-002c-43f8-9aff-d437c61d45c1.png)
![image](https://user-images.githubusercontent.com/73115616/135254745-97f2334b-25ed-4620-91ee-6b08bc4ea54f.png)

4. SSH into workspace container and see that there are folders `user-profile` and `user-preferences` on path `/config/user` containing files from secrets data
![image](https://user-images.githubusercontent.com/73115616/135255057-fbbbc860-c33e-4d11-ad23-d513cb28e457.png)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
